app/src/main/java/com/infomaniak/drive/ui/MainViewModel.kt

```diff
@@ -168,25 +168,31 @@ class MainViewModel(appContext: Application) : AndroidViewModel(appContext) {
         emit(ApiRepository.createOfficeFile(driveId, folderId, createFile))
     }
 
-    fun addFileToFavorites(file: File, userDrive: UserDrive? = null, onSuccess: (() -> Unit)? = null) = liveData(Dispatchers.IO) {
-        ApiRepository.postFavoriteFile(file).let { apiResponse ->
-            if (apiResponse.isSuccess()) {
-                FileController.updateFile(file.id, userDrive = userDrive) { localFile ->
-                    localFile.isFavorite = true
+    fun addFileToFavorites(file: File, userDrive: UserDrive? = null, onSuccess: (() -> Unit)? = null) {
+        liveData(Dispatchers.IO) {
+            with(ApiRepository.deleteFavoriteFile(file)) {
```

Comment:
Here we need to add to favorite, not delete

Reply:
Fixed.

Reply:
The issue has been fixed